src/Components/ServiceScene/RegexIcon.tsx

```diff
@@ -0,0 +1,45 @@
+import { useTheme2 } from '@grafana/ui';
+import React from 'react';
+import { GrafanaTheme2 } from '@grafana/data';
+import { css } from '@emotion/css';
+
+
+export type RegexInputValue = 'regex' | 'match'
+interface Props {
+  onRegexToggle: (state: RegexInputValue) => void;
+  regex: boolean;
+}
+
+export const RegexIcon = (props: Props) => {
+  const theme = useTheme2();
+  const fill = props.regex ? theme.colors.text.maxContrast : theme.colors.text.disabled;
+  const styles = getStyles(theme, fill);
+
+  return (
+    <span className={styles.container} title={`${props.regex ? 'String comparison' : 'Regex matching' }`}>
+      <svg
+        onClick={() => props.onRegexToggle(props.regex ? 'match' : 'regex')}
```

Comment:
Nitpicky, but having `onClick` listeners would make these not `Icon` but `IconButton`. 

Also we should probably enable accessibility rules for this repo, because elements with onClick listeners should be keyboard accessible, so this should be wrapped in a button. Same for LineFilterIcon.

Reply:
Good catch, should be fixed now